### PR TITLE
Create non www enforce middleware

### PR DESCRIPTION
**Pull request by karimmtarek**

### CODE REVIEW

**Comment by karimmtarek** [comment]
@bryanp can you please review and feedback, thanks.

File: pakyow-core/spec/unit/middleware/non_www_enforcer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        env['SERVER_NAME'] = host_www
+      end
+
+      it 'issues a 301 redirect and add www to host name' do
+        expect_any_instance_of(Pakyow::CallContext).to receive(:redirect).with(host, 301)
+        instance.call(env)
+      end
+    end
+
+    context 'www is not enforced and host doesn\'t have www' do
```
Comment (bryanp) [inline pakyow-core/spec/unit/middleware/non_www_enforcer_spec.rb]: Need a case for "www is enforced" that verifies the middleware isn't used.

**Comment by bryanp** [comment]
One comment on the specs, looks good otherwise.

**Comment by karimmtarek** [comment]
Done! Please review, thanks.

File: pakyow-core/spec/unit/middleware/non_www_enforcer_spec.rb [hunk 2]
```diff
@@ -37,6 +37,15 @@
     allow(app).to receive(:enforce_www)
     allow(Pakyow).to receive(:app).and_return(app)
     allow(app).to receive(:call)
+    Pakyow::App.instance_variable_set(:@builder, double(Rack::Builder).as_null_object)
+  end
+
+  describe 'www is enforced' do
+    before { allow(app).to receive(:enforce_www) { true } }
+
+   it 'does not use the non_www_enforcer middleware' do
```
Comment (bryanp) [inline pakyow-core/spec/unit/middleware/non_www_enforcer_spec.rb]: It looks like the indentation is off here. Picky, but can you fix?

**Comment by karimmtarek** [comment]
Sure, sorry for that.

File: pakyow-core/spec/unit/middleware/non_www_enforcer_spec.rb (hunk 2 above)
Comment (karimmtarek) [inline pakyow-core/spec/unit/middleware/non_www_enforcer_spec.rb]: Sorry for missing that, indentation fixed :)

**Comment by bryanp** [comment]
Thanks @karimmtarek!

**Pull request merged**